src/Core/Form/IdentifiableObject/DataHandler/CategoryFormDataHandler.php

```diff
@@ -101,17 +122,27 @@ public function create(array $data)
      */
     public function update($categoryId, array $data)
     {
+        $categoryId = (int) $categoryId;
+        $availableKeys = $this->menuThumbnailAvailableKeyProvider->getAvailableKeys($categoryId);
+
+        if (isset($data['menu_thumbnail_images']) && count($data['menu_thumbnail_images']) > count($availableKeys)) {
+            throw new MenuThumbnailsLimitException(sprintf('The maximum number of menu thumbnails has been reached for the %d category', $categoryId));
+        }
         $command = $this->createEditCategoryCommand($categoryId, $data);
 
         $this->commandBus->handle($command);
+        $categoryId = new CategoryId($categoryId);
 
-        $categoryId = new CategoryId((int) $categoryId);
+        /**
+         * In some cases in form menu_thumbnail_images can be disabled so value won't get here.
+         */
```

Comment:
style preferences... just a suggestion
```suggestion
        // In some cases in form menu_thumbnail_images can be disabled so value won't get here.
```

Reply:
I remember cs fixer complaining about smthing similar some time ago in one of my PR's :man_shrugging: